### PR TITLE
fix(server.js): terminate http server process gracefully

### PR DESCRIPTION
**Pull request by alepee**
On `SIGTERM`, `SIGINT` and `SIGQUIT`, node server should stop accepting new request
and exit when the last one already accepted has been processed.

### CODE REVIEW

**Comment by christophehurpeau** [comment]
@alepee why not use https://www.npmjs.com/package/http-graceful-shutdown ?

**Comment by alepee** [comment]
@christophehurpeau i juste gave it a try, works pretty well. @johangirod @LucasPerso could you also stress test those updates?

File: src/server/server.js [hunk 1]
```diff
@@ -78,10 +79,10 @@ if (process.env.NODE_ENV !== 'production' && module.hot) {
 const { port, host } = config.server;
 mountedServer.use(config.basePath, appServer());
 
-mountedServer.listen(process.env.PORT || port, process.env.HOST || host, () => {
+const server = mountedServer.listen(process.env.PORT || port, process.env.HOST || host, () => {
     readline.clearLine(process.stdout);
     readline.cursorTo(0, process.stdout);
-    process.stdout.write(`\x1b[0G${chalk.green('\u2713')} Server listening on: ${
+    process.stdout.write(`\x1b[0G${chalk.green('\u2713')} Server with PID ${process.pid} listening on: ${
```
Comment (johangirod) [inline src/server/server.js]: Maybe remove the PID process ?

**Comment by johangirod** [comment]
Seems good for me !

File: src/server/server.js [hunk 2]
```diff
@@ -92,3 +93,11 @@ mountedServer.listen(process.env.PORT || port, process.env.HOST || host, () => {
     }
 });
 
+httpGracefulShutdown(server, {
+    signals: 'SIGINT SIGTERM SIGQUIT',
+    timeout: 30000,
```
Comment (christophehurpeau) [inline src/server/server.js]: you don't use `timeout` argument ?

File: config/utils/killProcess.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+export default (process, { signal = 'SIGTERM', timeout = 15000 } = {}) => {
     if (!ALLOWED_SIGNALS.includes(signal)) {
         throw new Error(`Invalid signal: "${signal}", expecting one of ${ALLOWED_SIGNALS.join()}`);
     }
 
     return new Promise((resolve) => {
         const killTimeout = setTimeout(() => {
             // eslint-disable-next-line no-console
-            console.warn('[Vitamin] Server timeout: sending SIGKILL...');
+            console.warn(`[Vitamin] Process ${process.pid} timeout: sending SIGKILL...`);
```
Comment (johangirod) [inline config/utils/killProcess.js]: Still a PID ?

Comment (alepee) [inline config/utils/killProcess.js]: 🗡

Comment (LucasPerso) [inline config/utils/killProcess.js]: I am not really in favor of displaying a PID here. It might not be very user-friendly.

**Pull request merged**